### PR TITLE
nixos/mosh: add `programs.mosh.openFirewall`

### PR DESCRIPTION
**Pull request by Enzime**
## Description of changes

Add a `programs.mosh.openFirewall` option :)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/programs/mosh.nix [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
 in
 {
   options.programs.mosh = {
-    enable = mkOption {
-      description = lib.mdDoc ''
-        Whether to enable mosh. Note, this will open ports in your firewall!
-      '';
-      default = false;
-      type = lib.types.bool;
+    enable = lib.mkEnableOption "Whether to enable mosh.";
```
Comment (jian-lin) [inline nixos/modules/programs/mosh.nix:10]:
```suggestion
    enable = lib.mkEnableOption "mosh";
```

File: nixos/modules/programs/mosh.nix [hunk 2]
```diff
@@ -1,36 +1,29 @@
...
+    openFirewall = lib.mkEnableOption "Whether to automatically open the necessary ports in the firewall." // {
+      default = true;
```
Comment (jian-lin) [inline nixos/modules/programs/mosh.nix:12]:
```suggestion
    openFirewall = lib.mkEnableOption // {
      default = true;
      description = "Whether to open ports in the firewall.";
```

File: nixos/modules/programs/mosh.nix [hunk 3]
```diff
@@ -1,36 +1,29 @@
...
     };
-    withUtempter = mkOption {
-      description = lib.mdDoc ''
-        Whether to enable libutempter for mosh.
-        This is required so that mosh can write to /var/run/utmp (which can be queried with `who` to display currently connected user sessions).
-        Note, this will add a guid wrapper for the group utmp!
-      '';
+    withUtempter = lib.mkEnableOption (lib.mdDoc ''
```
Comment (jian-lin) [inline nixos/modules/programs/mosh.nix:14]:
```suggestion
    withUtempter = lib.mkEnableOption (''
```

File: nixos/modules/programs/mosh.nix [hunk 4]
```diff
@@ -1,36 +1,29 @@
...
     };
   };
 
-  config = mkIf cfg.enable {
-    environment.systemPackages = with pkgs; [ mosh ];
-    networking.firewall.allowedUDPPortRanges = [ { from = 60000; to = 61000; } ];
-    security.wrappers = mkIf cfg.withUtempter {
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.mosh ];
+    networking.firewall.allowedUDPPortRanges = lib.mkIf cfg.openFirewall [ { from = 60000; to = 61000; } ];
```
Comment (jian-lin) [inline nixos/modules/programs/mosh.nix:25]: `mkIf` is not needed here.

```suggestion
    networking.firewall.allowedUDPPortRanges = lib.optional cfg.openFirewall { from = 60000; to = 61000; };
```

File: nixos/modules/programs/mosh.nix [hunk 5]
```diff
@@ -1,36 +1,29 @@
...
+      default = true;
     };
-    withUtempter = mkOption {
-      description = lib.mdDoc ''
-        Whether to enable libutempter for mosh.
-        This is required so that mosh can write to /var/run/utmp (which can be queried with `who` to display currently connected user sessions).
-        Note, this will add a guid wrapper for the group utmp!
-      '';
+    withUtempter = lib.mkEnableOption (lib.mdDoc ''
+      Whether to enable libutempter for mosh.
```
Comment (jian-lin) [inline nixos/modules/programs/mosh.nix:15]: Same as above, your usage of `mkEnableOption` is wrong.

**Review by jian-lin — CHANGES_REQUESTED** [review]
What about factoring out changes unrelated to `openFirewall` into another commit?

File: nixos/modules/programs/mosh.nix (hunk 1 above)
Comment (Enzime) [reply nixos/modules/programs/mosh.nix]: Fixed :+1:

File: nixos/modules/programs/mosh.nix (hunk 3 above)
Comment (Enzime) [reply nixos/modules/programs/mosh.nix]: Fixed :+1:

File: nixos/modules/programs/mosh.nix (hunk 5 above)
Comment (Enzime) [reply nixos/modules/programs/mosh.nix]: Fixed :+1:

**Review by jian-lin — APPROVED** [review]

**Pull request merged**